### PR TITLE
OCPBUGS-7621-re: Fixed confusing Configuring an external LB section

### PR DESCRIPTION
**Pull request by dfitzmau**
[OCPBUGS-7621](https://issues.redhat.com/browse/OCPBUGS-7621)

Version(s):
4.12 and 4.11

Link to docs preview:
[Configuring an external load balancer](https://61966--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html#nw-osp-configuring-external-load-balancer_installing-vsphere-installer-provisioned-customizations)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/nw-osp-configuring-external-load-balancer.adoc [hunk 1]
```diff
@@ -29,9 +29,11 @@ to use an external load balancer in place of the default load balancer.
 
 * On your load balancer, port 22623, which is used to serve ignition startup configurations to nodes, is not exposed outside of the cluster.
 
-* Your load balancer must be able to access every machine in your cluster. Methods to allow this access include:
-** Attaching the load balancer to the cluster's machine subnet.
-** Attaching floating IP addresses to machines that use the load balancer.
+* Your load balancer can access the required ports on each node in your cluster. You can ensure this level of access by completing the following actions:
```
Comment (dfitzmau) [inline modules/nw-osp-configuring-external-load-balancer.adoc:32]: Matches 4.13 + docs.

File: modules/nw-configuring-external-load-balancer.adoc [hunk 2]
```diff
@@ -29,9 +27,11 @@ to use an external load balancer in place of the default load balancer.
...
-** Attaching the load balancer to the cluster's machine subnet.
-** Attaching floating IP addresses to machines that use the load balancer.
+* Your load balancer can access the required ports on each node in your cluster. You can ensure this level of access by completing the following actions:
++
+** The API load balancer can access ports `22623` and `6443` on the control plane nodes.
++
+** The ingress load balancer can access ports `443` and `80` on the nodes where the ingress pods are located.
 
 [IMPORTANT]
 ====
```
Comment (dfitzmau) [inline modules/nw-configuring-external-load-balancer.adoc:37]: SME query: Should I delete this IMPORTANT notice? 

Feedback on Jira:

> 2.  The below quoted section is confusing:

"""
[IMPORTANT]
====
External load balancing services and the control plane nodes must run on the same L2 network, and on the same VLAN when using VLANs to route traffic between the load balancing services and the control plane nodes.
====

**Comment by dfitzmau** [comment]
SME query: Feedback on Jira about health checks:

> This document does not reference our recommendations around Ingress Controller health checks: https://access.redhat.com/solutions/4677531

> Health checks are essential to ensuring that the external load balancer works as expected.  These should be included in the document.

We cannot add these links in our documentation, as they sometimes change or get deleted (depending on the nature of the knowledge-based article). Do you think mentioning health checks is needed? If so, maybe I could link to https://docs.openshift.com/container-platform/4.13/applications/application-health.html ? Is this section relevant?

File: modules/nw-configuring-external-load-balancer.adoc (hunk 2 above)
Comment (rvanderp3) [reply modules/nw-configuring-external-load-balancer.adoc:37]: yeah, that is a bit confusing.  i think it's fine to omit.

Comment (rvanderp3) [reply modules/nw-configuring-external-load-balancer.adoc:37]: otherwise LGTM

Comment (dfitzmau) [reply modules/nw-configuring-external-load-balancer.adoc:37]: I raised https://github.com/openshift/openshift-docs/pull/62749 to remove the note for all our versioned docs (4.14 through to 4.10)

**Comment by dfitzmau** [comment]
> SME query: Feedback on Jira about health checks:
> 
> > This document does not reference our recommendations around Ingress Controller health checks: https://access.redhat.com/solutions/4677531
> 
> > Health checks are essential to ensuring that the external load balancer works as expected.  These should be included in the document.
> 
> We cannot add these links in our documentation, as they sometimes change or get deleted (depending on the nature of the knowledge-based article). Do you think mentioning health checks is needed? If so, maybe I could link to https://docs.openshift.com/container-platform/4.13/applications/application-health.html ? Is this section relevant?

Health check content would be useful. Creating a new PR to tackle this work so that the doc updates can be transferred to all versioned vSphere doc sets.

**Comment by rvanderp3** [comment]
LGTM

**Review by kelbrown20 — COMMENTED** [review]
Hi Darragh! I checked the renders in all the sections and everything involving the content changes looks good to me. Great work! 
I did notice though, you said in the description that this pull request is for versions 4.12 and 4.11. Since you opened this PR off of 4.12, this PR will only apply to enterprise-4.12. We only cherrypick to a specific version when its off of main. If you want these changes to apply to 4.11 as well, you are going to have to open another PR off of the enterprise-4.11 branch.

**Comment by dfitzmau** [comment]
> Hi Darragh! I checked the renders in all the sections and everything involving the content changes looks good to me. Great work! I did notice though, you said in the description that this pull request is for versions 4.12 and 4.11. Since you opened this PR off of 4.12, this PR will only apply to enterprise-4.12. We only cherrypick to a specific version when its off of main. If you want these changes to apply to 4.11 as well, you are going to have to open another PR off of the enterprise-4.11 branch.

Hi @kelbrown20 . Thanks for letting me know. I did not know this. I'll create another PR for 4.11.

**Comment by dfitzmau** [comment]
/label merge-review-needed

I'll then create a new PR and cherry-pick the commit to my enterprise-4.11 equivalent.

File: modules/nw-configuring-external-load-balancer.adoc [hunk 3]
```diff
@@ -29,9 +27,11 @@ to use an external load balancer in place of the default load balancer.
 
 * On your load balancer, port 22623, which is used to serve ignition startup configurations to nodes, is not exposed outside of the cluster.
 
-* Your load balancer must be able to access every machine in your cluster. Methods to allow this access include:
-** Attaching the load balancer to the cluster's machine subnet.
-** Attaching floating IP addresses to machines that use the load balancer.
+* Your load balancer can access the required ports on each node in your cluster. You can ensure this level of access by completing the following actions:
```
Comment (stevsmit) [inline modules/nw-configuring-external-load-balancer.adoc:30]: Are these actual actions?

**Comment by stevsmit** [comment]
LGTM

**Comment by kelbrown20** [comment]
@dfitzmau My apologies Darragh, I thought we couldn't cherry pick from a one specific branch to another. That was my mistake, Thank you @stevsmit for merging!

**Pull request merged**